tests/run/line_trace.pyx

```diff
@@ -164,20 +166,27 @@ def cy_try_except(func):
     except KeyError as exc:
         raise AttributeError(exc.args[0])
 
+py_scope = {}
+exec("""
+def py_add(a,b):
+    x = a+b
+    return x
 
-def run_trace(func, *args, bint with_sys=False):
-    """
-    >>> def py_add(a,b):
-    ...     x = a+b
-    ...     return x
+def py_add_with_nogil(a,b):
+    x=a; y=b                     # 1
+    for _ in range(1):           # 2
+        z = 0                    # 3
+        z += py_add(x, y)        # 4
+    return z
 
-    >>> def py_add_with_nogil(a,b):
-    ...     x=a; y=b                     # 1
-    ...     for _ in range(1):           # 2
-    ...         z = 0                    # 3
-    ...         z += py_add(x, y)        # 4
-    ...     return z                     # 5
+def py_return(retval=123): return retval
+""", py_scope)
+py_add = py_scope['py_add']
+py_add_with_nogil = py_scope['py_add_with_nogil']
+py_return = py_scope['py_return']
```

Comment:
I think it would be better to do these assignments separately inside of the doctests to show where the functions are being used.

Also, let's not call the variable `py_scope` but `plain_python_functions`.